src/lib/libc_preload/gen_syscall_wrappers_c.py

```diff
@@ -66,27 +77,26 @@
     'sched_getaffinity',
     'sched_setaffinity',
     'select',
+    'setuid',
```

Comment:
> These wrapper functions (including the one for setuid()) employ a signal-based technique to ensure that when one thread changes credentials, all of the other threads in the process also change their credentials.

This one is kind of funny. It looks like the wrapper sends a signal to all threads, and the threads have signal handlers to change their uid.